### PR TITLE
- Add support for Python 3.12.  - Drop support for Python 3.7.

### PR DESCRIPTION
**Pull request by icemac**
Fix test in GHA.

### CODE REVIEW

**Comment by icemac** [comment]
Failed to pin `setuptools` to `< 69`: It gets ignored an `setuptools==70.1.0` is used for most Python versions. (Python 3.12 succeeds while using `setuptools==70.3.0`, which is also strange.)

So let's wait for https://github.com/buildout/buildout/pull/649 to be merged and released, which allows `setuptools >= 70`.

File: CHANGES.rst [hunk 1]
```diff
@@ -5,8 +5,13 @@ Change History
 3.1 (unreleased)
 ================
 
+- Add support for Python 3.12.
+
+- Drop support for Python 3.7.
+
 - Fix tests to run successfully against newer ``setuptools`` versions.
 
+- Pin ``setuptools < 69``, as newer versions break ``zc.buildout``.
```
Comment (dataflake) [inline CHANGES.rst:14]:
```suggestion
- Pin ``setuptools < 74``, as newer versions break ``zc.buildout``.
```

File: setup.py [hunk 2]
```diff
@@ -16,31 +16,25 @@ def read(*rnames):
...
-    python_requires='>=3.7',
+    python_requires='>=3.8',
     extras_require={
         'test': ['zope.testing'],
     },
     install_requires=[
         'zc.buildout >= 1.2.0',
         'zope.testrunner',
-        'setuptools',
+        'setuptools < 69',
```
Comment (dataflake) [inline setup.py:37]:
```suggestion
        'setuptools < 74',
```

**Review by dataflake — CHANGES_REQUESTED** [review]
What's with that setuptools pin in setup.py and why `< 69`?

**Review by dataflake — APPROVED** [review]

**Comment by dataflake** [comment]
I removed the pin (and change log entry for it) and everything's still green.

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**